core/micrometer/build.gradle.kts

```diff
@@ -21,12 +21,32 @@ val jupiterVersion: String by project
 val okHttpVersion: String by project
 val jodahFailsafeVersion: String by project
 val micrometerVersion: String by project
+val rsApi: String by project
 
 
 dependencies {
     api(project(":core:base"))
     api("com.squareup.okhttp3:okhttp:${okHttpVersion}")
     api("io.micrometer:micrometer-core:${micrometerVersion}")
+
+    testImplementation(project(":extensions:api:observability"))
+    testImplementation(testFixtures(project(":extensions:azure:azure-test")))
+    testImplementation(testFixtures(project(":launchers:junit")))
+    testImplementation(testFixtures(project(":common:util")))
+
+    runtimeOnly(project(":extensions:http:jersey-micrometer"))
+    runtimeOnly(project(":extensions:http:jetty-micrometer"))
+}
+
+tasks.withType<Test> {
+    val agent = rootDir.resolve("opentelemetry-javaagent.jar")
+    if (agent.exists()) {
+        jvmArgs(
+            "-javaagent:${agent.absolutePath}",
```

Comment:
In the integration test logs I see `[otel.javaagent 2022-03-21 10:08:37:351 +0000] [OkHttp http://localhost:4317/...] ERROR io.opentelemetry.exporter.internal.grpc.OkHttpGrpcExporter - Failed to export spans. The request could not be executed. Full error message: Failed to connect to localhost/0:0:0:0:0:0:0:1:4317`. Is this expected with this change?

Reply:
The spans can't be exported because the default exporter is grpc exporter, and there is no grpc server running to collect the metrics. When the spans can't be exporter, an error log appears to notify the user but it does not make the application fail on purpose.
This is "expected", because this test is only about metrics, so I didn't set up a mechanism to collect the spans.